### PR TITLE
check if project is bookmarked

### PR DESCRIPTION
**Pull request by jmilldotdev**
## What does this PR do and why?

_Provide a description of what this PR does. Link to any relevant GitHub issues, Notion tasks or Discord discussions._

## Screenshots or screen recordings

_If applicable, provide screenshots or screen recordings to demonstrate the changes._

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

**Comment by jmilldotdev** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2280** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2280" target="_blank">Graphite</a> 
    * **PR #2281** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2281" target="_blank">Graphite</a> 
      * **PR #2282** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2282" target="_blank">Graphite</a> 
        * **PR #2283** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2283" target="_blank">Graphite</a> 
          * **PR #2284** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2284" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2284?utm_source=stack-comment).

File: src/components/Project/ProjectHeader/BookmarkProjectButton.tsx [hunk 1]
```diff
@@ -29,14 +37,31 @@ const BookmarkProjectButton = () => {
     }
   }
 
+  const unbokmarkProject = async () => {
```
Comment (tomquirk) [inline src/components/Project/ProjectHeader/BookmarkProjectButton.tsx:40]:
```suggestion
  const unbookmarkProject = async () => {
```

File: src/components/Project/ProjectHeader/BookmarkProjectButton.tsx [hunk 2]
```diff
@@ -29,14 +37,31 @@ const BookmarkProjectButton = () => {
...
   return (
-    <Tooltip title={t`Bookmark Project`} placement="bottom">
-      <Button
-        onClick={() => bookmarkProject()}
-        icon={<StarOutlined />}
-        type="text"
-      />
-    </Tooltip>
+    <>
+      {isLoading && <></>}
```
Comment (tomquirk) [inline src/components/Project/ProjectHeader/BookmarkProjectButton.tsx:46]: **suggestion** maybe slap the loading icon from antd here

File: src/hooks/db/firebase/IsProjectBookmarked.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { ProjectMetadataContext } from 'contexts/projectMetadataContext'
+import { doc } from 'firebase/firestore'
+import { useWallet } from 'hooks/Wallet'
+import { useContext } from 'react'
+import { useDocumentData } from 'react-firebase-hooks/firestore'
+import { firestore } from 'utils/db/firebase/firebaseApp'
+
+export const useIsProjectBookmarked = () => {
```
Comment (tomquirk) [inline src/hooks/db/firebase/IsProjectBookmarked.ts:8]: **praise** this comes out so clean! love it.

**Review by tomquirk — APPROVED** [review]

File: src/components/Project/ProjectHeader/BookmarkProjectButton.tsx (hunk 2 above)
Comment (jmilldotdev) [reply src/components/Project/ProjectHeader/BookmarkProjectButton.tsx:64]: Can do this but imo would look a little weird to have a small loader in the actions bar.

Maybe the unfilled star icon if we're fetching?

**Pull request merged**